src/models/Steps.php

```diff
@@ -90,41 +93,63 @@ public function read(ContentParamsInterface $params): array
-    public function readAll(): array
+    public function readAll(string $table): array
     {
-        $sql = "SELECT experiments.id, experiments.title, stepst.finished, stepst.steps_body, stepst.steps_id
-            FROM experiments
-            CROSS JOIN (
-                SELECT item_id, finished,
-                GROUP_CONCAT(experiments_steps.body ORDER BY experiments_steps.ordering SEPARATOR '|') AS steps_body,
-                GROUP_CONCAT(experiments_steps.id ORDER BY experiments_steps.ordering SEPARATOR '|') AS steps_id
-                FROM experiments_steps
-                WHERE finished = 0 GROUP BY item_id) AS stepst ON (stepst.item_id = experiments.id)
-            WHERE userid = :userid GROUP BY experiments.id ORDER BY experiments.id DESC";
+        $whereClause = ' WHERE entity.userid = :userid';
+
+        $sql = $this->getReadAllSql($table, $whereClause);
 
         $req = $this->Db->prepare($sql);
         $req->bindParam(':userid', $this->Entity->Users->userData['userid'], PDO::PARAM_INT);
         $this->Db->execute($req);
 
         $res = $this->Db->fetchAll($req);
 
-        // clean up the results so we get a nice array with experiment id/title and steps with their id/body
-        // use reference to edit in place
-        foreach ($res as &$exp) {
-            $stepIDs = explode('|', $exp['steps_id']);
-            $stepsBodies = explode('|', $exp['steps_body']);
+        return $this->cleanUpReadAllSQLResults($res);
+    }
 
-            $expSteps = array();
-            foreach ($stepIDs as $key => $stepID) {
-                $expSteps[] = array($stepID, $stepsBodies[$key]);
-            }
-            $exp['steps'] = $expSteps;
-            unset($exp['steps_body'], $exp['steps_id'], $exp['finished']);
+    /**
+     * Get unfinished items steps from team
+     */
+    public function readAllTeam(): array
+    {
+        $teamgroupsOfUser = array_column((new TeamGroups($this->Entity->Users))->readGroupsFromUser(), 'id');
+        $teamgroups = '';
+        foreach ($teamgroupsOfUser as $teamgroup) {
+            $teamgroups .= " OR entity.canread = $teamgroup";
         }
 
-        return $res;
+        $whereClause = " WHERE entity.team = :teamid
+            AND (
+                entity.canread = 'public'
+                OR entity.canread = 'organization'
+                OR entity.canread = 'team'
+                $teamgroups
+                OR (entity.userid = :userid
+                    AND (
+                        entity.canread = 'user'
+                        OR entity.canread = 'useronly'
+                    )
+                )
+            )";
+
+        $sql = $this->getReadAllSql('items', $whereClause);
+
+        $req = $this->Db->prepare($sql);
+        $req->bindParam(':userid', $this->Entity->Users->userData['userid'], PDO::PARAM_INT);
+        $req->bindParam(':teamid', $this->Entity->Users->team, PDO::PARAM_INT);
+        $this->Db->execute($req);
+
+        $res = $req->fetchAll();
```

Comment:
you can now use `$this->Db->fetchAll($req);` and remove the check for false. See 3cd60633dc690efd0124af67d7ef1d8e192364bb